### PR TITLE
[WFLY-19846] Promote MicroProfile Telemetry 2.0 to WildFly Standard

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-19846

- Migrate MP Tel 2.0 changes to the WF feature pack
- Remove/clean up lingering otel/mptel changes in WF Preview 
- Remove temporary checks for WF Preview in the otel integration code

Resolves https://github.com/wildfly/wildfly-proposals/issues/619

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
IMHO we need CI setup for this branch to get these merged.

**Comment by kabir** [comment]
@jasondlee Failures seem related, so it's good we didn't merge before getting CI on it :-)

**Comment by rhusar** [comment]
@jasondlee Just FYI, both mine and Kabir's RFEs require this to work, it would help things a lot to have this passing so we can start building on top of it. Thanks!

**Comment by jasondlee** [comment]
> @jasondlee Just FYI, both mine and Kabir's RFEs require this to work, it would help things a lot to have this passing so we can start building on top of it. Thanks!

Yup. Actively working on it.

File: observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java [hunk 1]
```diff
@@ -56,22 +55,19 @@
...
+        "io.opentelemetry.api",
+        "io.opentelemetry.api.events",
+        "io.opentelemetry.context",
+        "io.opentelemetry.exporter",
+        "io.opentelemetry.otlp",
+        "io.opentelemetry.sdk",
+        "io.opentelemetry.semconv",
+        "io.smallrye.opentelemetry",
+        "io.opentelemetry.instrumentation.api",
+        "io.opentelemetry.semconv");
```
Comment (marekkopecky) [inline observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java:70]: This seems to be duplication with line 67, can you please double-check?

**Comment by rhusar** [comment]
Bleh! @kabir can you please sync [wildfly:mp-7-wip](https://github.com/wildfly/wildfly/tree/mp-7-wip) branch?

File: observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java (hunk 1 above)
Comment (jasondlee) [reply observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java:70]: It is. Good eye. Fixed.

File: testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/opentelemetry/BaseOpenTelemetryTest.java [hunk 2]
```diff
@@ -19,44 +19,46 @@
...
-            = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
-            + "<web-app xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://java.sun.com/xml/ns/javaee\"\n"
-            + "         xsi:schemaLocation=\"http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_3_0.xsd\"\n"
-            + "         metadata-complete=\"false\" version=\"3.0\">\n"
-            + "    <servlet-mapping>\n"
-            + "        <servlet-name>jakarta.ws.rs.core.Application</servlet-name>\n"
-            + "        <url-pattern>/*</url-pattern>\n"
-            + "    </servlet-mapping>"
-            + "</web-app>";
+        = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```
Comment (kabir) [inline testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/opentelemetry/BaseOpenTelemetryTest.java:27]: Not important but it is easier/tidier to include a class like this in your web app, which removes the need for this xml

```

import jakarta.ws.rs.ApplicationPath;
import jakarta.ws.rs.core.Application;

@ApplicationPath("/")
public class MyTestApplication extends Application {
}

```

Comment (jasondlee) [reply testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/opentelemetry/BaseOpenTelemetryTest.java:27]: Agreed. There's even one in a subpackage there. This is just an old wart that I haven't gotten around to cleaning up, but I'll get that on the next push.

**Pull request merged**